mne/io/nirx/nirx.py

```diff
@@ -513,3 +512,16 @@ def _convert_fnirs_to_head(trans, fro, to, src_locs, det_locs, ch_locs):
     det_locs = apply_trans(mri_head_t, det_locs)
     ch_locs = apply_trans(mri_head_t, ch_locs)
     return src_locs, det_locs, ch_locs, mri_head_t
+
+
+def _nirs_sort_idx(info):
```

Comment:
Could you please document here the desired sorting outcome.

Reply:
The idea is just to preserve the order on disk as much as possible, at least until we remove any restriction on the order,. I can document this but to me it seems like best (backward compatible as much as possible!) temporary compromise. Make sense? I can document this is the code but it just follows my suggestion in the previous PR comment

Reply:
But this PR suggests we apply this ordering to SNIRF files. And that is not preserving the ordering of the manufacturer. This seems relevant as more manufacturers use snirf, and we are pushing the NIRx ordering onto those files. So someone coming from a vendor that uses SNIRF may not know what the NIRx ordering is and will not expect it to follow NIRx disk order.

Reply:
The idea is just this PR applies a less restrictive reordering / tries harder to preserve the order on disk than a name based sort. If channels are already ordered in pairs -- even if it starts with S10_D9 and ends with S1_D1 -- it won't do any reordering. So in this sense the idea is to just to change less in the meantime until we remove any restriction whatsoever. Make sense?

Reply:
Sounds great. Thanks Eric